prov/efa/src/rdm/efa_rdm_ope.c

```diff
@@ -1191,7 +1194,9 @@ ssize_t efa_rdm_txe_prepare_local_read_pkt_entry_mr(struct efa_rdm_ope *txe)
 
 	assert(pkt_entry_copy->mr);
 	txe->local_read_pkt_entry = pkt_entry_copy;
-	txe->rma_iov[0].addr = (uint64_t)pkt_entry_copy->wiredata + pkt_offset;
```

Comment:
I'm a little confused what this original code was doing, it was setting ```txe->rma_iov[0].addr``` to ```(uint64_t)pkt_entry_copy->wiredata + pkt_offset```, but ```pkt_offset``` was ```(char *)txe->rma_iov[0].addr - pkt_entry->wiredata```

So it was setting ```txe->rma_iov[0].addr = (uint64_t)pkt_entry_copy->wiredata + (char *)txe->rma_iov[0].addr - pkt_entry->wiredata``` or ```txe->rma_iov[0].addr = (char *)txe->rma_iov[0].addr```

What exactly was going on here?

Reply:
`pkt_entry_copy` is a new packet (from read copy pool)  that copied the content in the old `pkt_entry`, 

But that pkt_offset should not change (`(char *)txe->rma_iov[0].addr - pkt_entry->wiredata`), it is the offset of the real application data in `pkt_entry->wiredata`, which also includes the pkt hdr.

So the old code does not do anything wrong, it was equivalent to `pkt_offset = rxr_pkt_req_data_offset(pkt_entry)`;

Reply:
Oh I see I misread the "copy" portion, it makes sense now. I thought it was pkt_entry->wiredata for everything